src/lib/logical_query_plan/stored_table_node.cpp

```diff
@@ -45,18 +90,48 @@ bool StoredTableNode::is_column_nullable(const ColumnID column_id) const {
 std::shared_ptr<TableStatistics> StoredTableNode::derive_statistics_from(
     const std::shared_ptr<AbstractLQPNode>& left_input, const std::shared_ptr<AbstractLQPNode>& right_input) const {
   DebugAssert(!left_input && !right_input, "StoredTableNode must be leaf");
-  return StorageManager::get().get_table(table_name)->table_statistics();
+
+  const auto stored_statistics = StorageManager::get().get_table(table_name)->table_statistics();
+
+  if (_pruned_column_ids.empty()) {
+    return stored_statistics;
+  }
+
+  /**
+   * Prune `_pruned_column_ids` from the statistics
```

Comment:
Is it correct that this ignores the pruned chunks? With the histogram PR, we should at least scale the statistics.

Reply:
Correct is relative when it comes to statistics ;) 
Yes, I guess there'd be something to be gained from adjusting statistics when pruning. Opened #1672